sds/src/secondary_validation.rs

```diff
@@ -122,6 +125,58 @@ impl Validator for GithubTokenChecksum {
     }
 }
 
+fn nhs_multiplier_from_number_idx(index: usize) -> u32 {
+    11 - ((index + 1) as u32)
+}
+
+impl Validator for NhsCheckDigit {
+    fn is_valid_match(&self, regex_match: &str) -> bool {
+        // https://www.datadictionary.nhs.uk/attributes/nhs_number.html
+        // The NHS number is a 10-digit number in the format 123 456 7890.
+        let stripped_match = regex_match
+            .chars()
+            .filter(|c| c.is_ascii_digit())
+            .collect::<String>();
+
+        if stripped_match.len() != 10 {
+            return false;
+        }
+        let mut total_sum = 0;
+        let mut check_digit = 0;
+        for (i, c) in stripped_match.chars().enumerate() {
```

Comment:
Here in the happy path, the code is going to go over `regex_match` 2 times (on to strip the regex, one to evaluate the content). Both can be done at the same time like in the `LuhnChecksum`. The gain will be marginal compared to other parts of the scanner but as it is a relatively simple change, I think it's worth doing it now (rather than having to come back on each checksum to optimise it later).

To do so, possibly `get_next_digit` can be extracted from the luhn checksum function.